MFVNeutralino/plugins/MovedTracksTreer.cc

```diff
@@ -143,8 +142,16 @@ void MFVMovedTracksTreer::analyze(const edm::Event& event, const edm::EventSetup
                 (*move_vertex)[1] - nt_filler.bs().y((*move_vertex)[2]),
                 (*move_vertex)[2]);
 
-    for (const reco::TrackRef& tk : *sel_tracks)
+    //for (const reco::TrackRef tk : *sel_tracks)
+    //  tks_push_back(*tk);
+
+    for (reco::TrackRef tk : *sel_tracks) {
+      const int whichtk = nt.tracks().n();
       tks_push_back(*tk);
+      auto vf = nt_filler.pvs_filler();
+      const int whichpv = nt_filler.tracks_filler().which_pv(event, &vf, tk);
+      nt.tracks().set_which_pv(whichtk, whichpv);
+    }
```

Comment:
Without following down to all of the definitions, I am surprised that `whichtk` is equal to `nt.tracks().n()` rather than an index. Isn't `n` a count of the number of tracks? Maybe I am missing something that would be clearer if I dug in.